### PR TITLE
Bgp fsmstate

### PR DESCRIPTION
**Pull request by SouravSinhaRoy**
This PR addresses addition of fsm_state in  BGP Metrics to address the issue in ( https://github.com/open-traffic-generator/models/issues/295 )
The redocli view of the changes is available @ https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/open-traffic-generator/models/bgp_fsmstate/artifacts/openapi.yaml&nocors#tag/Metrics.
Example snappi config and fetching of Metrics ( Note that this is a sample script) .The example config is a ebgp one with same as number,the bgp session will not come up and fsm state will oscillate between idle and active 

Go snappi example: 
```go
package gosnappi_test

import (
	"fmt"
	"log"
	"strings"
	"testing"

	"github.com/open-traffic-generator/snappi/gosnappi"
)

var mockGrpcServerLocation = "127.0.0.1:50001"
var mockHttpServerLocation = "http://127.0.0.1:50002"

func init() {
	if err := gosnappi.StartMockGrpcServer(mockGrpcServerLocation); err != nil {
		log.Fatal("Mock Grpc server init failed")
	}
	http_path := strings.Split(mockHttpServerLocation, "//")
	gosnappi.StartMockHttpServer(http_path[1])

}

func TestDevices(t *testing.T) {
	api := gosnappi.NewApi()
	api.NewGrpcTransport().SetLocation(mockGrpcServerLocation)
	config := api.NewConfig()
	device := config.Devices().Add().SetName("d1")
	device1 := config.Devices().Add().SetName("d2")

	eth := device.Ethernets().Add().
		SetPortName("p1").
		SetName("p1_eth1").
		SetMac("00:00:11:11:00:00").
		SetMtu(1500)
	eth.Ipv4Addresses().Add().SetName("p1_ip1").SetGateway("20.20.20.1").SetAddress("20.20.20.2")
	bgp := device.Bgp().SetName("p1_d1_bgp")
	bgp.SerRouterId("1.1.1.1")
	bgpiface1 = bgp.Ipv4Interfaces().Add()..SetIpv4Name("p1_ip1")
	bgppeer1 := 
       bgpiface1.peers()..v4peer().Add().SetName("p1_bgp1").SetPeerAddress("20.20.20.1").SetAsType("ebgp").SetAsNumber("100")

	eth1 := device1.Ethernets().Add().
		SetPortName("p2").
		SetName("p2_eth1").
		SetMac("00:00:12:11:00:00").
		SetMtu(1500)
	eth1.Ipv4Addresses().Add().SetName("p2_ip1").SetGateway("20.20.20.2").SetAddress("20.20.20.1")
	bgp1 := device1.Bgp().SetName("p2_d1_bgp")
	bgp1.SerRouterId("20.20.20.20")
	bgpiface2 = bgp1.Ipv4Interfaces().Add()..SetIpv4Name("p2_ip1")
	bgppeer2 := 
       bgpiface2.peers()..v4peer().Add().SetName("p2_bgp1").SetPeerAddress("20.20.20.2").SetAsType("ebgp").SetAsNumber("100")
	fmt.Println(config.ToJson())

	api.SetConfig(config)
       // bgp fsm state fetch
	dNames := []string{}
	for _, d := range config.Devices().Items() {
		bgp := d.Bgp()
		for _, ip := range bgp.Ipv4Interfaces().Items() {
			for _, peer := range ip.Peers().Items() {
				dNames = append(dNames, peer.Name())
			}
		}
	}
	dMetrics, err := client.GetBgpv4Metrics(dNames, nil)
	if err != nil {
		return false, err
	}
	var sessionsUp int32 = 0
	var sessionsIdleState int32 = 0
	var sessionsActiveState int32 = 0

	for _, d := range dMetrics.Items() {
		if d.FsmState() == gosnappi.Bgpv4MetricFsmState.IDLE {
			sessionsIdleState += 1
		}
		if d.FsmState() == gosnappi.Bgpv4MetricFsmState.ACTIVE {
			sessionsActiveState += 1
		}
		if d.SessionState() == gosnappi.Bgpv4MetricSessionState.UP {
			sessionsUp += 1
		}
	}
}
```

Python example: 
```python
import otg
#Creating Ports
api = otg.api(location="https://localhost")
config = api.config()
p1 = config.ports.port(name='p1', location='172.17.0.2:50071')[-1]
p2 = config.ports.port(name='p2', location='172.17.0.3:50071')[-1]

#Create BGP running on connected interface port 1.
p1_d1 = config.devices.device(name='p1_d1')[-1]

p1_eth1 = p1_d1.ethernets.ethernet(port_name=p1.name)[-1]
p1_eth1.name = 'p1_eth1'
p1_eth1.mac = '64:00:00:00:00:01'
p1_ip1 = p1_eth1.ipv4_addresses.ipv4(name = 'p1_ip1', address = '20.20.20.2', gateway = '20.20.20.1')[-1]

p1_d1.bgp.router_id = '1.1.1.1'
p1_bgpv4Iface = p1_d1.bgp.ipv4_interfaces.v4interface(ipv4_name=p1_ip1.name)[-1]
p1_bgpPeer1 = p1_bgpv4Iface.peers.v4peer(name='p1_peer1', peer_address='20.20.20.1', as_type='ebgp', as_number=100)[-1]

p2_d1 = config.devices.device(name='p2_d1')[-1]

p2_eth1 = p2_d1.ethernets.ethernet(port_name=p2.name)[-1]
p2_eth1.name = 'p2_eth1'
p2_eth1.mac = '65:00:00:00:00:01'
p2_ip1 = p1_eth1.ipv4_addresses.ipv4(name = 'p2_ip1', address = '20.20.20.1', gateway = '20.20.20.2')[-1]

p2_d1.bgp.router_id = '2.2.2.2'
p2_bgpv4Iface = p2_d1.bgp.ipv4_interfaces.v4interface(ipv4_name=p2_ip1.name)[-1]
p2_bgpPeer1 = p2_bgpv4Iface.peers.v4peer(name='p2_peer1', peer_address='20.20.20.2', as_type='ebgp', as_number=100)[-1]
print(config.serialize())
```

### CODE REVIEW

**Review by SuryyaKrJana — APPROVED** [review]
Looks fine

File: result/bgpv4.yaml [hunk 1]
```diff
@@ -144,3 +146,27 @@ components:
...
           type: integer
           x-field-uid: 15
+        fsm_state:
+          description: >-
+            Bgp peer fsm state as idle, connect, active, opensent, openconfirm and established. In all the states except Established the bgp session is down,
+            Idle refers to the Idle state of the FSM.
+            Connect refers to the state where the session is waiting for the underlying transport session to be established.
+            Active refers to the state where the session is awaiting for a connection from the remote peer.
+            OpenSent refers to the state where the session is in the process of being established. The local system has sent an OPEN message.
+            OpenConfirm refers to the state where the session is in the process of being established. The local system is awaiting a NOTIFICATION or KEEPALIVE message from remote peer.
```
Comment (apratimmukherjee) [inline result/bgpv4.yaml:156]: local system has sent and received an OPEN message and is awaiting ...

**Review by apratimmukherjee — COMMENTED** [review]
Need to provide example for OTG stat access and remove GNMI stat access in provided example.

File: result/bgpv4.yaml (hunk 1 above)
Comment (SouravSinhaRoy) [reply result/bgpv4.yaml]: updated

**Review by apratimmukherjee — APPROVED** [review]
Looks good now.

**Pull request merged**